test/components/popup.spec.tsx

```diff
@@ -0,0 +1,269 @@
+import React = require('react');
+import ReactDOM = require('react-dom');
+import {selectDom} from 'test-drive';
+import { ClientRenderer, expect, waitFor } from 'test-drive-react';
+import {PopupDemo} from '../../demo/components/popup-demo';
+import {HorizontalPosition, Popup, VerticalPosition} from '../../src/components/';
+
+const popup = 'POPUP';
+const container = 'POPUP_DEMO_DIV';
+
+describe('<Popup />', function() {
+    const clientRenderer = new ClientRenderer();
+    const bodySelect = selectDom(document.body);
+    const testContainer = document.body.appendChild(document.createElement('div'));
+
+    afterEach(function() {clientRenderer.cleanup(); });
+    after(() => {document.body.removeChild(testContainer); });
+
+    describe('The popup user', function() {
+        it('clicks on the parent and the popup opens and closes after another click', async function() {
+            const {select, waitForDom} = clientRenderer.render(<PopupDemo />);
+
+            await waitForDom(() => {
+                expect(select(container)).to.be.present();
+                expect(select(container, popup)).to.be.absent();
+            });
+            (select(container) as HTMLDivElement).click();
+            await waitForDom(() => {
+                expect(bodySelect(popup)).to.be.present();
+            });
+            (select(container) as HTMLDivElement).click();
+            return waitForDom(() => {expect(bodySelect(popup)).to.be.absent(); });
+        });
+    });
+
+    it('displays the popup and renders its children if the open prop is given', function() {
+        const div = document.body.appendChild(testContainer);
+
+        clientRenderer.render(
+            <Popup
+                anchor={div!}
+                open={true}
+            >
+                <span data-automation-id="SPAN">Popup Body</span>
+            </Popup>);
+
+        return waitFor(() => {
+            expect(bodySelect(popup)).to.be.present();
+            expect(bodySelect(popup, 'SPAN')).to.be.present();
+        });
+    });
+
+    it('removes the component when unmounting', async function() {
+        const div = document.body.appendChild(testContainer);
+
+        clientRenderer.render(
+            <Popup
+                anchor={div!}
+                open={true}
+            >
+                <span data-automation-id="SPAN">Popup Body</span>
+            </Popup>);
+
+        await waitFor(() => {expect(bodySelect(popup)).to.be.present(); });
+        ReactDOM.unmountComponentAtNode(bodySelect(popup)!.parentElement!);
```

Comment:
this is not how we do it.
You should create a rendering container in the test. Pass it to clientRenderer.render (second parameter), and only then you are allowed to unmount it yourself.
You should verify the current code does not crash outside the test (unmounting an already unmounted component).